test/metamodelutil.js

```diff
@@ -100,6 +101,31 @@ describe('MetaModel (Car)', () => {
     });
 });
 
+describe('MetaModel (with Maps & Scalars)', () => {
+    const modelManager = new ModelManager();
+
+    // Add base
+    const baseModelPath = path.resolve(__dirname, './cto/base.cto');
+    let baseModel = fs.readFileSync(baseModelPath, 'utf8');
+    modelManager.addCTOModel(baseModel, 'base.cto');
+
+    // Add Model which imports from base
+    const modelPath = path.resolve(__dirname, './cto/model.cto');
+    let model = fs.readFileSync(modelPath, 'utf8');
+    modelManager.addCTOModel(model, 'model.cto');
+
+    // The ModelFile resolved
+    const mapImportsResolved = JSON.parse(fs.readFileSync(path.resolve(__dirname, './cto/mapImportsResolved.json'), 'utf8'));
+
+    describe('#toMetaModel', () => {
+        it('should resolve all namespaces on a Model containing Map Types, where the Map Types are imported', async () => {
+            const modelFile = modelManager.getAst();
```

Comment:
You could avoid the dependency on concerto-core entirely by storing the AST JSON directly.